tests/bluetooth/tester/src/btp_vcp.c

```diff
@@ -0,0 +1,544 @@
+
+	net_buf_simple_init(buf, 0);
+	net_buf_simple_add_u8(buf, AICS_READ_SUPPORTED_COMMANDS);
+	net_buf_simple_add_u8(buf, AICS_INIT);
+	net_buf_simple_add_u8(buf, AICS_SET_GAIN);
+	net_buf_simple_add_u8(buf, AICS_MUTE);
+	net_buf_simple_add_u8(buf, AICS_UNMUTE);
+	net_buf_simple_add_u8(buf, AICS_MAN_GAIN);
+	net_buf_simple_add_u8(buf, AICS_AUTO_GAIN);
+	net_buf_simple_add_u8(buf, AICS_MAN_GAIN_ONLY);
+	net_buf_simple_add_u8(buf, AICS_AUTO_GAIN_ONLY);
+	net_buf_simple_add_u8(buf, AICS_DESCRIPTION);
+
+	tester_send(BTP_SERVICE_ID_AICS, AICS_READ_SUPPORTED_COMMANDS,
+		    CONTROLLER_INDEX, buf->data, buf->len);
+}
+
+static void aics_state_cb(struct bt_aics *inst, int err, int8_t gain,
+			  uint8_t mute, uint8_t mode)
+{
+	LOG_DBG("AICS state callback");
+}
+
+static void aics_gain_setting_cb(struct bt_aics *inst, int err, uint8_t units,
+				 int8_t minimum, int8_t maximum)
+{
+	LOG_DBG("AICS gain setting callback");
+}
+
+static void aics_input_type_cb(struct bt_aics *inst, int err,
+			       uint8_t input_type)
+{
+	LOG_DBG("AICS input type callback");
+}
+
+static void aics_status_cb(struct bt_aics *inst, int err, bool active)
+{
+	LOG_DBG("AICS status callback");
+}
+
+static void aics_description_cb(struct bt_aics *inst, int err,
+				char *description)
+{
+	LOG_DBG("AICS description callback");
+}
+
+static struct bt_aics_cb aics_cb = {
+	.state = aics_state_cb,
+	.gain_setting = aics_gain_setting_cb,
+	.type = aics_input_type_cb,
+	.status = aics_status_cb,
+	.description = aics_description_cb
+};
+
+void aics_set_gain(uint8_t *data, uint8_t len)
+{
+	const struct aics_set_gain *cmd = (void *) data;
+	uint32_t gain;
+	int err;
+
+	gain = sys_le32_to_cpu(cmd->gain);
+
+	LOG_DBG("AICS set gain 0x%04x", gain);
+
+	for (int i = 0; i < (int)CONFIG_BT_VCP_VOL_REND_AICS_INSTANCE_COUNT; i++) {
+		err = bt_aics_gain_set(included.aics[0], gain);
+		if (err) {
+			tester_rsp(BTP_SERVICE_ID_AICS, AICS_SET_GAIN,
+				   CONTROLLER_INDEX, BTP_STATUS_FAILED);
+		}
+	}
+}
+
+void aics_mute(void)
+{
+	int err;
+
+	LOG_DBG("AICS mute");
+
+	for (int i = 0; i < (int)CONFIG_BT_VCP_VOL_REND_AICS_INSTANCE_COUNT; i++) {
+		err = bt_aics_mute(included.aics[i]);
+		if (err) {
+			tester_rsp(BTP_SERVICE_ID_AICS, AICS_MUTE,
+				   CONTROLLER_INDEX, BTP_STATUS_FAILED);
+		}
+	}
+}
+
+void aics_unmute(void)
+{
+	int err;
+
+	LOG_DBG("AICS unmute");
+
+	for (int i = 0; i < (int)CONFIG_BT_VCP_VOL_REND_AICS_INSTANCE_COUNT; i++) {
+		err = bt_aics_unmute(included.aics[i]);
+		if (err) {
+			tester_rsp(BTP_SERVICE_ID_AICS, AICS_UNMUTE,
+				   CONTROLLER_INDEX, BTP_STATUS_FAILED);
+		}
+	}
+}
+
+void aics_man_gain(void)
+{
+	int err;
+
+	LOG_DBG("AICS manual gain set");
+
+	for (int i = 0; i < (int)CONFIG_BT_VCP_VOL_REND_AICS_INSTANCE_COUNT; i++) {
+		err = bt_aics_manual_gain_set(included.aics[i]);
+		if (err) {
+			tester_rsp(BTP_SERVICE_ID_AICS, AICS_MAN_GAIN,
+				   CONTROLLER_INDEX, BTP_STATUS_FAILED);
+		}
+	}
+}
+
+void aics_auto_gain(void)
```

Comment:
so how success is reported? btp handler returns void and all responses must be handled by callbacks

Reply:
Done